FlexMeter.c

```diff
@@ -0,0 +1,217 @@
+/*
+htop - FlexMeter.c
+(C) 2024 Stoyan Bogdanov
+Released under the GNU GPLv2+, see the COPYING file
+in the source distribution for its full text.
+*/
+
+#include <dirent.h>
+#include <pwd.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <time.h>
+#include <unistd.h>
+
+#include "CRT.h"
+#include "config.h"
+#include "FlexMeter.h"
+#include "Object.h"
+
+
+#define FLEX_CFG_FOLDER ".config/htop/FlexMeter"
+
+typedef struct {
+   char* name;
+   char* command;
+   char* type;
+   char* caption;
+   char* uiName;
+} _flex_meter;
+
+_flex_meter meter_list[METERS_LIST_SIZE];
+
+static int meters_count = 0;
+
+static const int DateMeter_attributes[] = {
+   FLEX
+};
+
+MeterClass* FlexMeter_class = NULL;
+
+static int check_for_meters(void);
+
+static bool parse_input(_flex_meter *meter, char* line)
+{
+   switch(line[0])
+   {
+      case 'n':
+         if (String_startsWith(line, "name=")) {
+            xAsprintf(&meter->uiName, "Flex: %s", line + 5);
+         }
+         break;
+      case 'c':
+         if (String_startsWith(line, "command=")) {
+            meter->command = xStrdup(line + 8);
+         } else if (String_startsWith(line, "caption=")) {
+            meter->caption = xStrdup(line + 8);
+         }
+         break;
+      case 't':
+         if (String_startsWith(line, "type=")) {
+            meter->type = xStrdup(line + 6);
+         }
+         break;
+      default:
+         return false;
+   }
+
+   return true;
+}
+
+static bool load_config(_flex_meter *meter, char* file)
+{
+   bool ret = false;
+   FILE* fp = fopen(file, "r");
+
+   if (fp != NULL) {
```

Comment:
Note, that `fclose(NULL);` is undefined behaviour, usually characterized to crash your program. ;-)

```suggestion
   if (!fp) {
      return false;
   }

```

Doing it that way around also allows to lower the indentation level.